pkg/client/cache/listers.go

```diff
@@ -151,11 +151,11 @@ type storeToNodeConditionLister struct {
 }
 
 // List returns a list of nodes that match the conditions defined by the predicate functions in the storeToNodeConditionLister.
-func (s storeToNodeConditionLister) List() (nodes api.NodeList, err error) {
+func (s storeToNodeConditionLister) List() (nodes []*api.Node, err error) {
```

Comment:
I think I'm OK with this. It was originally this way to try and match the client interface but I think it's already diverged, so I'd say go for it.